persistence/pg_test.go

```diff
@@ -152,3 +152,107 @@ func TestConcurrentHtlcSettle(t *testing.T) {
+				PaymentPreimage: lntypes.Preimage{2},
+				Value:           100,
+				PaymentAddr:     [32]byte{2},
+			},
+			htlcs: map[types.HtlcKey]int64{{Node: nodeKey, ChanID: 10, HtlcID: 12}: 70},
+		},
+		{
+			InvoiceCreationData: types.InvoiceCreationData{
+				PaymentPreimage: lntypes.Preimage{3},
+				Value:           100,
+				PaymentAddr:     [32]byte{2},
+			},
+			htlcs: map[types.HtlcKey]int64{{Node: nodeKey, ChanID: 10, HtlcID: 13}: 70},
+		},
+	}
+
+	for _, invoice := range invoices {
+		require.NoError(t, persister.RequestSettle(context.Background(),
+			&InvoiceCreationData{invoice.InvoiceCreationData},
+			invoice.htlcs,
+		))
+	}
+
+	// Settled the two first invoices but not the last
+	for i := 0; i < 2; i++ {
+		var invoiceSettled bool
+		for htlc := range invoices[i].htlcs {
+			var err error
+			invoiceSettled, err = persister.MarkHtlcSettled(context.Background(), htlc)
+			require.NoError(t, err)
+		}
+		require.True(t, invoiceSettled)
+	}
+
+	// Ask only for the first settled invoice
+	dbInvoices, err := persister.GetSettledInvoices(context.Background(), 1, 0)
+	require.NoError(t, err)
+	require.Len(t, dbInvoices, 1)
+	require.Equal(t, dbInvoices[0].SequenceNum, uint64(1))
+	require.Equal(t, dbInvoices[0].Preimage, invoices[0].PaymentPreimage)
+
+	// Ask for invoices with a sequence number greater than 1 (only the second invoice should match)
+	dbInvoices, err = persister.GetSettledInvoices(context.Background(), 1, 1)
+	require.NoError(t, err)
+	require.Len(t, dbInvoices, 1)
+	require.Equal(t, dbInvoices[0].SequenceNum, uint64(2))
+	require.Equal(t, dbInvoices[0].Preimage, invoices[1].PaymentPreimage)
+
+	// Ask only for all settled invoices
+	// It should be only 2 invoices in the list as the last one in not settled
+	dbInvoices, err = persister.GetSettledInvoices(context.Background(), 3, 0)
+	require.NoError(t, err)
+	require.Len(t, dbInvoices, 2)
+	require.Equal(t, dbInvoices[0].SequenceNum, uint64(1))
+	require.Equal(t, dbInvoices[0].Preimage, invoices[0].PaymentPreimage)
+	require.Equal(t, dbInvoices[1].SequenceNum, uint64(2))
+	require.Equal(t, dbInvoices[1].Preimage, invoices[1].PaymentPreimage)
+
+	// Settle the last invoice
+	var invoiceSettled bool
+	for htlc := range invoices[2].htlcs {
+		var err error
+		invoiceSettled, err = persister.MarkHtlcSettled(context.Background(), htlc)
+		require.NoError(t, err)
+	}
+	require.True(t, invoiceSettled)
+
+	// Now we should have 3 invoices
+	dbInvoices, err = persister.GetSettledInvoices(context.Background(), 3, 0)
+	require.NoError(t, err)
+	require.Len(t, dbInvoices, 3)
+	require.Equal(t, dbInvoices[0].SequenceNum, uint64(1))
+	require.Equal(t, dbInvoices[0].Preimage, invoices[0].PaymentPreimage)
+	require.Equal(t, dbInvoices[1].SequenceNum, uint64(2))
+	require.Equal(t, dbInvoices[1].Preimage, invoices[1].PaymentPreimage)
+	require.Equal(t, dbInvoices[2].SequenceNum, uint64(3))
+	require.Equal(t, dbInvoices[2].Preimage, invoices[2].PaymentPreimage)
+}
```

Comment:
Maybe a nice touch to add an assertion to `mux_test.go` where we query settled invoices. To give it a bit more of an e2e character.